src/typingsInstaller/nodeTypingsInstaller.ts

```diff
@@ -29,13 +29,14 @@ namespace ts.server.typingsInstaller {
     }
 
     /** Used if `--npmLocation` is not passed. */
-    function getDefaultNPMLocation(processName: string) {
+    function getDefaultNPMLocation(processName: string, host: InstallTypingHost): string {
         if (path.basename(processName).indexOf("node") === 0) {
-            return `"${path.join(path.dirname(process.argv[0]), "npm")}"`;
-        }
-        else {
-            return "npm";
+            const npmPath = path.join(path.dirname(process.argv[0]), "npm");

```

Comment:
Isnt this normally directory ? (and that resolving to `npm.cmd` or other depending on os) ?

Reply:
I don't understand. This was an extract constant of the existing value

Reply:
Eg. on windows using eg: `"c:\Program Files\nodejs\npm"` works because there is `"c:\Program Files\nodejs\npm.cmd"`. In your case the fileExists is going to fail because `"c:\Program Files\nodejs\npm"` isn't file but directory.

Reply:
I'm assuming there is always a npm executable based on usage. There seems to be one on windows and MacOs when I checked. If there is just npm dir, wouldn't running `/Users/matb/path/to/npm install ata-stuff` from the command line always fail?

Reply:
Oops my bad when I looked up I thought `npm` is folder and not file..